src/spaceapi_client.rs

```diff
@@ -135,27 +121,46 @@ fn fetch_binary(url :&String) -> Result<Vec<u8>,io::Error> {
   let mut v = vec![];
   let size = try!(res.read_to_end(&mut v));
   
-  Ok(v)
+  if size > 0 {
+    Ok(v)
+  } else {
+    Err(io::Error::new(io::ErrorKind::Interrupted, format!("fetch_binary({}) empty response", url)))
+  }
+}
 
-  /*match client.get(url).send() {
-    Err(e) => Err(format!("client.get({}) error:\nError: {}", url, e)),
-    Ok(mut res) => {
-      
-      match res.read_to_end(&mut v) {
-        Err(e) => Err(format!("")),
-        Ok(size) => {
-          Ok( v )
+fn extract_people_now_present(status : Optional<Sensors>) -> Result<String, String> {
+  match status { // FIXME why is this clone needed here?
+    Absent => Err(format!("response contains no sensors")),
+    Value(sensors) => {
+      match sensors.people_now_present {
+        Absent => Err(format!("response contains no sensors.people_now_present")),
+        Value(sensors) => {
+          if sensors.is_empty() {
+            Err(format!("response.sensors.people_now_present is empty"))
+          } else {
+            let mut r = "".into();
```

Comment:
Just to make a point,

.into() MIR:

```
    bb0: {
        tmp1 = const ""; // Scope(6) at <anon>:2:21: 2:23
        tmp0 = &(*tmp1); // Scope(6) at <anon>:2:21: 2:23
        var0 = <&str as std::convert::Into<std::string::String>>::into(tmp0) -> [return: bb3, unwind: bb2]; // Scope(5) at <anon>:2:21: 2:30
    }
```

Constructor MIR:

```
    bb3: {
        var1 = std::string::String::new() -> [return: bb5, unwind: bb4]; // Scope(9) at <anon>:3:13: 3:26
    }
```

https://is.gd/wDVub9 :wink: 

But into is also perfectly fine.